File: A-SOURCE_CODE/A-SERVICE/Shared/ViewModels/CommentReports/FindCommentReportsViewModel.cs
using Shared.Models;

namespace Shared.ViewModels.CommentReports
{
    public class FindCommentReportsViewModel
    {
        /// <summary>
        /// Comment report index.
        /// </summary>
        public int? Id { get; set; }

        /// <summary>
        /// Index of comment which should be reported.
        /// </summary>
        public int? CommentIndex { get; set; }

        /// <summary>
        /// Index of comment owner.
        /// </summary>
        public int? CommentOwnerIndex { get; set; }

        /// <summary>
        /// Index of person who created comment report.
        /// </summary>
        public int? CommentReporterIndex { get; set; }
        
        /// <summary>
        /// Body of comment which is reported.
        /// </summary>
        public TextSearch Body { get; set; }

        /// <summary>
        /// Reason of report.
        /// </summary>
        public TextSearch Reason { get; set; }

        /// <summary>
        /// When the comment should be created.
        /// </summary>
        public UnixDateRange Created { get; set; }

        /// <summary>
        /// Pagination information.
        /// </summary>
        public Pagination Pagination { get; set; }
    }
}

File: A-SOURCE_CODE/B-CLIENT/A-WEB/Administration/app/viewmodels/comment-report/FindCommentReportsViewModel.ts
import {UnixDateRange} from "../UnixDateRange";
import {Pagination} from "../Pagination";
import {TextSearch} from "../TextSearch";
import {CommentReportSortProperty} from "../../enumerations/order/CommentReportSortProperty";
import {SortDirection} from "../../enumerations/SortDirection";
import {TextSearchMode} from "../../enumerations/TextSearchMode";
export class FindCommentReportsViewModel{

    // Index of comment report.
    public id: number;

    // Index of comment.
    public commentIndex: number;

    // Index of comment owner.
    public ownerIndex: number;

    // Index of comment reporter.
    public reporterIndex: number;

    // Body of comment.
    public body: TextSearch;

    // Reason of comment.
    public reason: TextSearch;

    // Property which should be used for results sorting.
    public sort: CommentReportSortProperty;

    // Direction of sorting.
    public direction: SortDirection;

    // Time when the comment was created.
    public created: UnixDateRange;

    // Results pagination.
    public pagination: Pagination;

    // Initiate filter with default settings.
    public constructor(){

        let commentBody = new TextSearch();
        commentBody.mode = TextSearchMode.contains;
        commentBody.value = '';
        this.body = commentBody;

        let commentReportReason = new TextSearch();
        commentReportReason.mode = TextSearchMode.contains;
        commentReportReason.value = '';
        this.reason = commentReportReason;

        this.sort = CommentReportSortProperty.Index;
        this.direction = SortDirection.Ascending;
        this.pagination = new Pagination();
    }
}